contract/models/contract.py

```diff
@@ -213,9 +213,35 @@ def action_contract_send(self):
         }
 
     @api.model
-    def _get_recurring_create_invoice_domain(self, contract=False):
+    def _finalize_invoice_values(self, invoice_values):
+        # If no account on the product, the invoice lines account is
+        # taken from the invoice's journal in _onchange_product_id
+        # This code is not in finalize_creation_from_contract because it's
+        # not possible to create an invoice line with no account
+        new_invoice = self.env['account.invoice'].new(invoice_values)
+        for invoice_line in new_invoice.invoice_line_ids:
+            invoice_line.invoice_id = new_invoice
+            invoice_line._onchange_product_id()
+        return new_invoice._convert_to_write(new_invoice._cache)
+
+    @api.model
+    def _finalize_invoice_creation(self, invoices):
+        for invoice in invoices:
+            invoice._onchange_partner_id()
+        invoices.compute_taxes()
+
+    @api.model
+    def _get_contracts_to_invoice_domain(self, date_ref=None, contract=None):
+        """
+        This method builds the domain to use to find all
+        contracts (account.analytic.account) to invoice.
+        :param date_ref: optional reference date to use instead of today
+        :param contract: optional contract (account.analytic.account) to target
+        :return: list (domain) usable on account.analytic.account
+        """
         domain = []
-        date_ref = fields.Date.context_today(self)
+        if not date_ref:
+            date_ref = fields.Date.context_today(self)
```

Comment:
commenting about the lines *below*, it looks we have no use case for the contract parameter anymore. If so, remove it.

Reply:
done